flaskbb/templates/macros.html

```diff
@@ -308,12 +308,52 @@
     {%- endif -%}
 {% endmacro %}
 
-{% macro navlink(endpoint, name, icon='', active='') %}
+{% macro navlink(endpoint, name, icon='', active='', urlforkwargs=None) %}
 <li {% if endpoint == request.endpoint or endpoint == active or active == True %}class="active"{% endif %}>
-    <a href="{{ url_for(endpoint) }}">{% if icon %}<i class="{{ icon }}"></i> {% endif %} {{ name }}</a>
+    <a href="{% if urlforkwargs %}{{ url_for(endpoint, **urlforkwargs) }}{% else %}{{ url_for(endpoint) }}{% endif %}">
+        {% if icon %}<i class="{{ icon }}"></i> {% endif %} {{ name }}
+    </a>
+</li>
+{% endmacro %}
+
+{% macro externalnavlink(uri, name, icon='') %}
+<li>
+    <a href="{{uri}}">{% if icon %}<i class="{{ icon }}"></i> {% endif %} {{ name }}</a>
 </li>
 {% endmacro %}
 
+{% macro navtext(text, icon="", cls="") %}
+<li{% if cls %} class="{{ cls }}"{% endif %}><a href="#">{% if icon %}<i class="{{ icon }}"></i> {% endif %}{{ text }}</a></li>
```

Comment:
@sh4nks halp. Tried using `.navbar-text` but it acts weird, making them inline instead of stacked. I'm not crazy about wrapping text items in empty `<a>` tags.

Reply:
Do we have an example for the navtext? What is the navtext used for?

Reply:
Hm. I thought I just one, but header is probably enough.